src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4288,14 +4288,173 @@ private static bool UseLegacyWarnings(BoundExpression expr)
 
         public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
         {
-            // https://github.com/dotnet/roslyn/issues/29618: Assign each of the deconstructed values,
-            // and handle deconstruction conversion for node.Right.
-            VisitLvalue(node.Left);
-            VisitRvalue(node.Right.Operand);
+            var left = node.Left;
+            var variables = GetDeconstructionAssignmentVariables(left);
+
+            // PROTOTYPE: Should visit expressions on right side, even if errors.
+            if (!node.HasErrors)
+            {
+                var right = node.Right;
+                VisitDeconstructionArguments(variables, right.Conversion, right.Operand);
+            }
+
+            variables.FreeAll(v => v.NestedVariables);
             SetResult(node);
             return null;
         }
 
+        private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> variables, Conversion conversion, BoundExpression right)
```

Comment:
Local function?

Reply:
It seemed too long for a local function.

---
In reply to: [252750155](https://github.com/dotnet/roslyn/pull/32994#discussion_r252750155) [](ancestors = 252750155)